controllers/datadogagent/agent.go

```diff
@@ -394,6 +407,198 @@ func buildAgentNetworkPolicy(dda *datadoghqv1alpha1.DatadogAgent, name string) *
 	return policy
 }
 
+func (b agentNetworkPolicyBuilder) PodSelector() metav1.LabelSelector {
+	return metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			kubernetes.AppKubernetesInstanceLabelKey: datadoghqv1alpha1.DefaultAgentResourceSuffix,
+			kubernetes.AppKubernetesPartOfLabelKey:   NewPartOfLabelValue(b.dda).String(),
+		},
+	}
+}
+
+func (b agentNetworkPolicyBuilder) ddFQDNs() []cilium.FQDNSelector {
+	selectors := []cilium.FQDNSelector{}
+
+	ddURL := b.dda.Spec.Agent.Config.DDUrl
+	if ddURL != nil {
+		selectors = append(selectors, cilium.FQDNSelector{
+			MatchName: strings.TrimPrefix(*ddURL, "https://"),
+		})
+	}
+
+	var site string
+	if b.dda.Spec.Site != "" {
+		site = b.dda.Spec.Site
+	} else {
+		site = defaultSite
+	}
+
+	selectors = append(selectors, []cilium.FQDNSelector{
+		{
+			MatchPattern: fmt.Sprintf("*-app.agent.%s", site),
+		},
+		{
+			MatchName: fmt.Sprintf("agent-intake.logs.%s", site),
+		},
+		{
+			MatchName: fmt.Sprintf("process.%s", site),
+		},
+		{
+			MatchName: fmt.Sprintf("orchestrator.%s", site),
+		},
+	}...)
+
+	return selectors
+}
+
+func (b agentNetworkPolicyBuilder) BuildCiliumPolicy() *cilium.NetworkPolicy {
+	specs := []cilium.NetworkPolicySpec{
+		{
+			Description:      "Egress to ECS agent port 51678",
+			EndpointSelector: b.PodSelector(),
+			Egress: []cilium.EgressRule{
+				{
+					ToCIDR: []string{"169.254.0.0/16"},
+					ToPorts: []cilium.PortRule{
+						{
+							Ports: []cilium.PortProtocol{
+								{
+									Port:     "51678",
+									Protocol: cilium.ProtocolTCP,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
```

Comment:
We might need to retrofit here DataDog/helm-charts#391 and to add
```yaml
ToEntities: []cilium.Entity{
	cilium.EntityHost,
},
```
to fix DataDog/helm-charts#387 also with the operator.